src/ajax/formats.cljc

```diff
@@ -65,40 +63,41 @@
    ;;; Must be a format generating function
    :else (format-entry request)))
 
-(m/defn-curried get-accept-entries [request format-entry]
+(defn get-accept-entries [request format-entry]
   (let [fe (if (vector? format-entry)
              (first format-entry)
              (:content-type (get-format request format-entry)))]
     (cond (nil? fe) ["*/*"]
           (string? fe) [fe]
           :else fe)))
 
-(m/defn-curried content-type-matches
+(defn content-type-matches
   [^String content-type ^String accept]
   (or (= accept "*/*")
       (>= (.indexOf content-type accept) 0)))
 
-(m/defn-curried detect-content-type
+(defn detect-content-type
   [content-type request format-entry]
   (let [accept (get-accept-entries request format-entry)]
-    (some (content-type-matches content-type) accept)))
+    (some #(content-type-matches content-type %) accept)))
 
 (defn get-default-format
   [response {:keys [response-format] :as request}]
-  (let [f (detect-content-type (u/get-content-type response) request)]
+  (let [f #(detect-content-type (u/get-content-type response) request %)]
     (->> response-format
          (filter f)
          first
          (get-format request))))
 
-(m/defn-curried detect-response-format-read
-  [request response]
-  (let [format (get-default-format response request)]
-    ((:read format) response)))
+(defn detect-response-format-read
+  [request]
+  (fn [response]
+    (let [format (get-default-format response request)]
+      ((:read format) response))))
 
 (defn accept-header [{:keys [response-format] :as request}]
   (if (vector? response-format)
-    (mapcat (get-accept-entries request) response-format)
+    (mapcat #(get-accept-entries request %) response-format)
```

Comment:
Not really related to your change, but I'm wondering if I should have written

(let [formats (if (vector? response-format) response-format [response-format]))]
   mapcat &c)

Reply:
This makes sense; I rewrote it like this.